fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/runtime/MPJWTAuthExtensionArchivePreparer.java

```diff
@@ -119,35 +119,44 @@ public void process() throws Exception {
         String publicKey = fraction.getPublicKey();
         if (publicKey != null) {
             log.debugf("PublicKey: %s", publicKey);
-
-            if (publicKey.startsWith("file:")) {
-                File fileRef = new File(publicKey.substring(5, publicKey.length()));
-                war.addAsManifestResource(new FileAsset(fileRef), "MP-JWT-SIGNER");
-            } else if (publicKey.startsWith("classpath:")) {
-                String cpref  = publicKey.substring(10, publicKey.length());
-                Node node = archive.get("WEB-INF/classes/" + cpref);
-                war.addAsManifestResource(node.getAsset(), "MP-JWT-SIGNER");
+            if (publicKey.startsWith("file:") || publicKey.startsWith("classpath:")) {
+                log.warn("Using 'thorntail.microprofile.jwt.token.signer-pub-key' for the 'file:' or 'classpath:' key "
+                        + "assets is deprecated, use the 'thorntail.microprofile.jwt.token.signer-pub-key-location' "
+                        + "property instead");
+                if (publicKey.startsWith("file:")) {
+                    File fileRef = new File(publicKey.substring(5, publicKey.length()));
+                    war.addAsManifestResource(new FileAsset(fileRef), "MP-JWT-SIGNER");
+                } else if (publicKey.startsWith("classpath:")) {
+                    war.addAsManifestResource(new StringAsset(publicKey), "MP-JWT-SIGNER-KEY-LOCATION");
+                }
             } else {
                 war.addAsManifestResource(new StringAsset(publicKey), "MP-JWT-SIGNER");
             }
         }
 
+        String publicKeyLocation = fraction.getPublicKeyLocation();
+        if (publicKeyLocation != null) {
+            log.debugf("PublicKey location: %s", publicKeyLocation);
+            war.addAsManifestResource(new StringAsset(publicKeyLocation), "MP-JWT-SIGNER-KEY-LOCATION");
+        }
+
         war.addAsManifestResource(new StringAsset("" + fraction.getExpGracePeriodSecs().get()), "MP-JWT-EXP-GRACE");
 
         if (fraction.isDefaultMissingMethodPermissionsDenyAccess()) {
             war.addAsManifestResource(EmptyAsset.INSTANCE, "MP-JWT-DENY-NONANNOTATED-METHODS");
         }
 
         if (fraction.getJwksUri() != null) {
+            log.warn("Using 'thorntail.microprofile.jwt.token.jwks-uri' for the HTTPS based JWK sets is deprecated, "
+                    + "use the 'thorntail.microprofile.jwt.token.signer-pub-key-location' "
+                    + "property instead");
             log.debugf("JwksUri: %s", fraction.getJwksUri());
-            war.addAsManifestResource(new StringAsset(fraction.getJwksUri()), "MP-JWT-JWKS");
+            war.addAsManifestResource(new StringAsset(fraction.getJwksUri()), "MP-JWT-SIGNER-KEY-LOCATION");
```

Comment:
Should we only add `MP-JWT-SIGNER-KEY-LOCATION` if one doesn't exist? This code would mean that `thorntail.microprofile.jwt.token.jwks-uri` would override `thorntail.microprofile.jwt.token.signer-pub-key-location`, if both are specified.

(This becomes quite complex already...)

Reply:
I think may be I should ignore the jwksUri value if the location is set, but the same could be said if the signer-pub-key is set. Agree it is not becoming simpler, given that it is only a deprecation so there is need to support overlapping properties. I don't really mind much even abandoning this PR (though the current state is not close to the perfection either :-) ), let me tweak few things and then see how sound it is.

Reply:
Yea, agree, overlapping properties are more of a corner case.